net/base/io_buffer.cc

```diff
@@ -113,7 +113,15 @@ void GrowableIOBuffer::SetCapacity(int capacity) {
   // this will get reset in `set_offset`.
   data_ = nullptr;
   // realloc will crash if it fails.
+  // Calling reallocate with size 0 and a non-null pointer causes memory leaks
```

Comment:
In c25 these customizations were not wrapped with macros, I assume because we were thinking about upstreaming this code. I'm planning to add macro guards here though, as we've been doing it even with code we may upstream